lib/mpl_toolkits/mplot3d/tests/test_axes3d.py

```diff
@@ -2276,6 +2276,24 @@ def test_on_move_vertical_axis(vertical_axis: str) -> None:
     )
 
 
+@pytest.mark.parametrize(
+    "vertical_axis, aspect_expected",
+    [
+        ("x", [1.190476, 0.892857, 1.190476]),
+        ("y", [0.892857, 1.190476, 1.190476]),
+        ("z", [1.190476, 1.190476, 0.892857]),
+    ],
+)
+def test_set_box_aspect_vertical_axis(vertical_axis, aspect_expected):
+    ax = plt.subplot(1, 1, 1, projection="3d")
+    ax.view_init(elev=0, azim=0, roll=0, vertical_axis=vertical_axis)
+    ax.figure.canvas.draw()
+
+    ax.set_box_aspect(None)
+
+    np.testing.assert_allclose(aspect_expected, ax._box_aspect)
```

Comment:
```suggestion
    np.testing.assert_allclose(aspect_expected, ax._box_aspect, rtol=1e-6)
```
Should fix the issue.